src/main/java/utils/CustomFileReaderFactory.java

```diff
@@ -0,0 +1,13 @@
+package utils;
+
+import utils.filereaders.CsvFileReader;
+
+public class CustomFileReaderFactory {
+    public CustomFileReader getFileReader(String extension) {
+        if (extension.equals("csv")) {
```

Comment:
instead of if else,
do CustomFileReader fileReader =  CUSTOM_FILE_READER.get(fileType), rename extension to fileType
if fileReader == null throw exception
else return fileReader